app/models/manageiq/providers/ibm_power_hmc/inventory/parser/infra_manager.rb

```diff
@@ -33,6 +34,37 @@ def parse_cecs
     end
   end
 
+  def parse_cecs_unavailable
+    collector.cecs_unavailable.each do |sys|
+      mtype_model, serial = sys["MTMS"].split("*")
+      host = persister.hosts.build(
+        :uid_ems             => sys["UUID"],
+        :ems_ref             => sys["UUID"],
+        :name                => sys["SystemName"],
+        :hypervisor_hostname => sys["MTMS"],
+        :ipaddress           => sys["IPAddress"],
+        :power_state         => lookup_power_state(sys["State"]),
+        :vmm_vendor          => "ibm_power_hmc",
+        :type                => ManageIQ::Providers::IbmPowerHmc::InfraManager::Host.name
```

Comment:
A. This should be built automatically by the persister builder auto_sti_class method
B. If you do have to manually provider a type name you can just use the string `"ManageIQ::Providers::IbmPowerHmc::InfraManager::Host"` going to the `class.name` is just overhead

```suggestion
```

Reply:
Ok, I have removed the line for both available and unavailable hosts.